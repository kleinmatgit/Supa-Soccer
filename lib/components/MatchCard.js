import React from 'react';
import PropTypes from 'prop-types';
import TeamName from 'helpers/TeamName';
import TeamImage from 'components/TeamImage';

function displayScore(score) {
  return score === null ? '-' : score;
}

class MatchCard extends React.Component {

  isWinner = (homeOrAway) => {
    let homeScore = this.props.match.score.fullTime.homeTeam;
    let awayScore = this.props.match.score.fullTime.awayTeam;
    return ((homeOrAway === 'home' && homeScore > awayScore)
      || (homeOrAway === 'away' && homeScore < awayScore) );
  }

  getClassNameScore = (homeOrAway) => {
    let cls = 'match-card-scores';
    if(this.isWinner(homeOrAway)) {
      cls += ' score-winner';
    }
    return cls;
  }

  getClassNameTeam = (homeOrAway) => {
    let cls = 'match-card-teams';
    if(this.isWinner(homeOrAway)) {
      cls += ' team-winner';
    }
    return cls;
  }

  formatDate = () => {
    var weekday = new Array(7);
    weekday[0] =  'Sunday';
    weekday[1] = 'Monday';
    weekday[2] = 'Tuesday';
    weekday[3] = 'Wednesday';
    weekday[4] = 'Thursday';
    weekday[5] = 'Friday';
    weekday[6] = 'Saturday';

    var d = new Date(this.props.match.utcDate),
      day = weekday[d.getDay()].substring(0,3),
      date = d.getDate(),
      month = d.getMonth();
    return (day + ', ' + date + '/' + month);
  }

  // formatTime = (dateString) => {
  //   var d = new Date(dateString), day =
  //   return (day + ', ' + date + '/' + month);
  // }

  render() {

    let homeTeam = TeamName.getDisplayName(this.props.match.homeTeam.name);
    let awayTeam = TeamName.getDisplayName(this.props.match.awayTeam.name);

    return (
      <div className="match-card">
        <table>
          <tbody>
            <tr>
              <td>
                <table className="match-card-table">
                  <tbody>
                    <tr>
                      <td className="match-card-crest"><TeamImage team={this.props.match.homeTeam.name} /></td>
                      <td className={this.getClassNameTeam('home')}>{homeTeam}</td>
                      <td className={this.getClassNameScore('home')}>{displayScore(this.props.match.score.fullTime.homeTeam)}</td>
                    </tr>
                    <tr>
                      <td className="match-card-crest"><TeamImage team={this.props.match.awayTeam.name} /></td>
                      <td className={this.getClassNameTeam('away')}>{awayTeam}</td>
                      <td className={this.getClassNameScore('away')}>{displayScore(this.props.match.score.fullTime.awayTeam)}</td>
                    </tr>
                  </tbody>
                </table>
              </td>
              <td>
                <div className="match-card-details">
                  <div className="match-card-details-status">{this.props.match.status}</div>
                  <div className="match-card-details-date">{this.formatDate()}</div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}

MatchCard.propTypes = {
  match: PropTypes.object.isRequired
};

export default MatchCard;
