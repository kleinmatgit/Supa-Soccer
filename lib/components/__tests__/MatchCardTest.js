import React from 'React';
import MatchCard from 'components/MatchCard';
import renderer from 'react-test-renderer';

describe('MatchCard component', () => {

  const testProps = {
    match: {
      'id': 233039,
      'utcDate': '2018-08-18T11:30:00Z',
      'status': 'FINISHED',
      'matchday': 2,
      'homeTeam': {
        'id': 715,
        'name': 'Cardiff City FC'
      },
      'awayTeam': {
        'id': 67,
        'name': 'Newcastle United FC'
      },
      'score': {
        'winner': 'DRAW',
        'duration': 'REGULAR',
        'fullTime': {
          'homeTeam': 0,
          'awayTeam': 0
        },
        'halfTime': {
          'homeTeam': 0,
          'awayTeam': 0
        },
        'extraTime': {
          'homeTeam': null,
          'awayTeam': null
        },
        'penalties': {
          'homeTeam': null,
          'awayTeam': null
        }
      }
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <MatchCard {...testProps}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
