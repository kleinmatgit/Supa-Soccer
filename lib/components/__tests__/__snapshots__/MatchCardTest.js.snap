// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchCard component renders correctly 1`] = `
<div
  className="col-sm-6"
  id="match-card-container"
>
  <div
    className="match-card"
  >
    <table>
      <tbody>
        <tr>
          <td>
            <div
              className="match-card-score-table"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      <img
                        className="team-img"
                        src="images/cardiff-city.png"
                      />
                    </td>
                    <td
                      className="match-card-team"
                    >
                      Cardiff City
                    </td>
                    <td
                      className=""
                    >
                      0
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <img
                        className="team-img"
                        src="images/newcastle.png"
                      />
                    </td>
                    <td
                      className="match-card-team"
                    >
                      Newcastle
                    </td>
                    <td
                      className=""
                    >
                      0
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </td>
          <td>
            <div
              className="match-card-status-table"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      FT
                    </td>
                  </tr>
                  <tr>
                    <td
                      className=" datetime-match-finished"
                    >
                      Sat, 18/7
                    </td>
                  </tr>
                  <tr>
                    <td
                      className=" datetime-match-finished"
                    >
                      19:30
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
